src/policy/protect_payload.rs

```diff
@@ -110,15 +113,26 @@ impl PolicyModule for ProtectPayloadPolicy {
             }
         }
 
-        // TODO: Work on that section
-        // Step 2: Restore sensitives privileged registers
-        /*self.write_confidential_registers(ctx, true);*/
+        if self.first_jump {
+            // Lock memory from all cores
+            broadcast_policy_interrupt();
+        }
```

Comment:
Here you need a proper barrier, because there is no guarantee that the interrupt will be received immediately  (and it practice it wont, there are delays).
If you don't wait for an acknowledgement from all of the cores you might be in a state where the payload is protected only on some cores and thus can be attacked from the remaining unlocked cores.

The simple solution to create a single usage barrier is to have a static atomic counter, you set it once to the number of cores, then each core decrease it by one and wait in a loop for the count to reach 0. You can add a [`core::hint::spin_loop`](https://doc.rust-lang.org/stable/core/hint/fn.spin_loop.html) in the spin loop to calm down the CPU while spinning.